runtime/server/assets.go

```diff
@@ -0,0 +1,52 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/rilldata/rill/runtime/pkg/httputil"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func (s *Server) assetsHandler(w http.ResponseWriter, req *http.Request) error {
+	ctx := req.Context()
+	instanceID := req.PathValue("instance_id")
+	path := req.PathValue("path")
+
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.instance_id", instanceID),
+		attribute.String("args.path", path),
+	)
+
+	if !auth.GetClaims(req.Context()).CanInstance(instanceID, auth.ReadObjects) {
+		return httputil.Errorf(http.StatusForbidden, "does not have access to assets")
+	}
+
+	repo, release, err := s.runtime.Repo(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	paths := repo.GetCachedPaths()
+	allowed := false
+	for _, p := range paths {
```

Comment:
It's too implicit to rely on the repo's cached paths as an allow list. It only works as long as that feature is only used for `PublicPaths`. 

Instead, I would suggest explicitly adding `PublicPaths` as a property on `Instance`, assigning it in `UpdateInstanceWithRillYAML`, and checking it here.

Reply:
so you suggest adding an instance reference to repo, right? because there's no instance in the params list 
![image](https://github.com/rilldata/rill/assets/14215045/ae445c52-1412-4886-9015-7476b644426b)

Reply:
No, it could just pass the paths to cache, no need to pass the full instance. But as discussed above, it might be fine just to cache the whole repo on a persistent disk.

Reply:
I would still suggest to omit caching from this PR since it's not so important